### DISCUSSION TITLE
How to change icons/color of .js files under "test" folder?

### DISCUSSION THREAD

**Discussion by mcntai** [mallowigi/a-file-icon-idea#252] [Q&A]
Hello,
I'm using Webstorm 2021.3

I'm using "One Dark Theme" as my theme and Atom Material Icons plugin for the icons.

I'd like to have .js files which are under "test" folder to be read. Right now all js files are the same and it's not really convenient.

Is there a way to configure it?

**Comment by mallowigi** [comment]
Unfortunately there is no support for "paths", since we are looking inside files specifically. I can add a feature to look for a path in the url but its not planned for the moment.

Also I highly invite you to NOT use this pattern. As a JS advocate, I am always pushing for putting tests alongside source files, just saying.

Another possibility for you would be to add a suffix to your test files, such as `blabla.test.js` or `blabla.spec.js`. We already have specific icons for those.

**Reply by mcntai** [reply]
Thanks a lot for your response!

If you are looking inside files, shouldn't it see some keywords like "describe", "it" inside test files and make such files look different compared to other js files?

It's not my local project, it's work project where a lot of test are written already and I can't change the suffix extensions for all these files either.

**Reply by mallowigi** [reply]
Nope, you can't parse folders easily. I suppose you could do it since JB does it for PSI Icons but it would be overkill for a small project like this one

**Reply by mcntai** [reply]
Forget about "test" folder I mentioned in the beginning. 

You said, "we are looking inside files specifically". So when you are looking inside .js files which contain words like "describe", "it", "before" and maybe something else, do you make such file to look like a file with tests or it will look like the other .js files ?

**Comment by mallowigi** [comment]
No, you misunderstood, I meant we are looking into files, as in the file system, as opposed to their absolute paths. Currently we don't look for their contents.

Again, I think it could be possible but that would be overkill as there are plenty of languages and frameworks, and doing so for all of them would be a nightmare while usually having the right extension would be sufficient in most cases.

--
Elior

**Comment by mallowigi** [comment]
PS: this has been made available since then.
